File: android/src/main/java/com/degica/komoju/android/sdk/types/Language.kt
package com.degica.komoju.android.sdk.types

import java.util.Locale

enum class Language(val languageCode: String) {
    ENGLISH(languageCode = "en"),
    JAPANESE(languageCode = "ja"),
    ;

    internal fun toLocale(): Locale = when (this) {
        ENGLISH -> Locale.ENGLISH
        JAPANESE -> Locale.JAPANESE
    }
    companion object {
        val default
            get() = when (Locale.getDefault().language) {
                Locale.JAPANESE.language -> JAPANESE
                else -> ENGLISH
            }
    }
}

File: komoju-ios-sdk/KomojuSDK/types/Language.swift
import Foundation

public enum Language {
    case english
    case japanese

    public var languageCode: String {
        switch self {
        case .english: return "en"
        case .japanese: return "ja"
        }
    }
}

public extension Language {
    var useSystemLanguage: Language {
        return if Locale.current.languageCode == "en" {
            .english
        } else {
            self
        }
    }

    func parse(from string: String) -> Language {
        return switch string {
        case "en": .english
        case "ja": .japanese
        default: useSystemLanguage
        }
    }
}
